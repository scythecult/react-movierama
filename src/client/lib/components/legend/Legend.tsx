import { Seat } from '../seat/Seat';
import styles from './styles.module.css';

export const Legend = () => {
  return (
    <ul className={styles.legend}>
      <li className={styles.legendItem}>
        <Seat isDisabled />
        <small>N/A</small>
      </li>
      <li className={styles.legendItem}>
        <Seat isDisabled isSelected />
        <small>Selected</small>
      </li>
      <li className={styles.legendItem}>
        <Seat isOccupied />
        <small>Occupied</small>
      </li>
    </ul>
  );
};
