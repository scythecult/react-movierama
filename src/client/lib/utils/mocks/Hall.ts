import type { SeatNode } from './SeatNode';

type HallOptions = {
  Seat: typeof SeatNode;
  hallPlanMatrix: number[][];
  seatWidth: number;
  seatHeight: number;
  seatOffset: number;
};

export class Hall {
  #hallPlanMatrix: number[][];
  #seatWidth: number;
  #seatHeight: number;
  #seatOffset: number;
  #canvasWidth = 0;
  #canvasHeight = 0;
  #finalRowLength = 0;
  #Seat: typeof SeatNode;
  #hall: SeatNode[] = [];

  constructor(options: HallOptions) {
    this.#Seat = options.Seat;
    this.#hallPlanMatrix = options.hallPlanMatrix;
    this.#seatWidth = options.seatWidth;
    this.#seatHeight = options.seatHeight;
    this.#seatOffset = options.seatOffset;
  }

  createSeats() {
    for (let i = 0; i < this.#hallPlanMatrix.length; i++) {
      const currentRowLength = this.#hallPlanMatrix[i].length - 1;

      for (let j = 0; j < this.#hallPlanMatrix[i].length; j++) {
        const seatMatrixValue = this.#hallPlanMatrix[i][j];
        const isGap = !seatMatrixValue;

        if (this.#finalRowLength < currentRowLength) {
          this.#finalRowLength = currentRowLength;
        }

        this.#canvasWidth = this.#finalRowLength * (this.#seatWidth + this.#seatOffset) + this.#seatWidth;
        this.#canvasHeight = i * (this.#seatHeight + this.#seatOffset) + this.#seatHeight;

        if (!isGap) {
          const seatXOffset = isGap
            ? this.#seatWidth + this.#seatOffset + this.#seatWidth / 2
            : this.#seatWidth + this.#seatOffset;
          const x = j * seatXOffset;
          const y = i * (this.#seatHeight + this.#seatOffset);

          this.#hall.push(
            new this.#Seat({
              id: x + y,
              x,
              y,
              place: j + 1,
              type: seatMatrixValue,
              row: i + 1,
            }),
          );
        }
      }
    }

    return this.#hall;
  }

  getCanvasSize() {
    if (!this.#canvasWidth || !this.#canvasHeight) {
      this.createSeats();
    }

    return {
      width: this.#canvasWidth,
      height: this.#canvasHeight,
    };
  }
}
