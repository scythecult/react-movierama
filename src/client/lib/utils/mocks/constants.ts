// TODO Should move to server
export const SEAT_WIDTH = 40;

export const SEAT_HEIGHT = 34;

export const SEAT_OFFSET = 10;

export const HALL_PLAN_MATRIX =
  // HALL
  [
    // ROW 1 - seat 0 - empty space width === 1 seat
    [1, 1, 0, 1, 4, 4, 1, 0, 1, 1, 1, 1, 1],
    // ROW
    [2, 2, 0, 1, 1, 1, 1, 0, 1],
    // ROW
    [1, 1, 0, 2, 2, 2, 2, 0, 1],
    // ROW
    [1, 1, 0, 3, 3, 3, 3, 0, 1],
  ];
