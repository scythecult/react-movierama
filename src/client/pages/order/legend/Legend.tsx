import type { SeatType } from '@/client/lib/types/OrderData';
import { Seat } from '../seats/seat/Seat';
import styles from './styles.module.css';

export type LegendProps = {
  seatTypes: SeatType[];
};

export const Legend = (props: LegendProps) => {
  const { seatTypes } = props;

  const seatTypeNodes = seatTypes.map((seatType) => {
    const { id, name, ticketTypes } = seatType;
    const [defaultTicketType] = ticketTypes;

    return <li className={styles.legendItem} key={id}>
      <Seat className={styles.legendItemSeat} type={name} />
      <div className={styles.legendItemWrap}>
        <small className={styles.legendItemName}>{name}</small>
        <small className={styles.legendItemPrice}>{defaultTicketType.price} RUB</small>
      </div>
    </li>;
  });

  return (
    <ul className={styles.legend}>
      {seatTypeNodes}
      <li className={styles.legendItem}>
        <Seat className={styles.legendItemSeat} isDisabled />
        <small className={styles.legendItemName}>N/A</small>
      </li>
      <li className={styles.legendItem}>
        <Seat className={styles.legendItemSeat} isDisabled isSelected />
        <small className={styles.legendItemName}>Selected</small>
      </li>
      <li className={styles.legendItem}>
        <Seat className={styles.legendItemSeat} isOccupied />
        <small className={styles.legendItemName}>Occupied</small>
      </li>
    </ul>
  );
};
