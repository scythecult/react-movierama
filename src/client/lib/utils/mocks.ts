import type { SeatData, SeatType } from '@/client/lib/types/OrderPageData';
import { HALL_PLAN_MATRIX, SEAT_HEIGHT, SEAT_OFFSET, SEAT_WIDTH } from '@/client/lib/utils/mocks/constants';
import { Hall } from '@/client/lib/utils/mocks/Hall';
import { SeatNode } from '@/client/lib/utils/mocks/SeatNode';

export const getMockSeatTypes = (): SeatType[] => {
  return [
    {
      id: 1,
      name: 'regular',
      ticketTypes: [
        {
          id: 1,
          name: 'adult',
          price: 330,
        },
        {
          id: 2,
          name: 'child',
          price: 165,
        },
      ],
    },
    {
      id: 2,
      name: 'comfort',
      ticketTypes: [
        {
          id: 1,
          name: 'adult',
          price: 380,
        },
        {
          id: 2,
          name: 'child',
          price: 190,
        },
      ],
    },
    {
      id: 3,
      name: 'vip',
      ticketTypes: [
        {
          id: 1,
          name: 'adult',
          price: 430,
        },
        {
          id: 2,
          name: 'child',
          price: 165,
        },
      ],
    },
    {
      id: 4,
      name: 'wheelchair',
      ticketTypes: [
        {
          id: 1,
          name: 'adult',
          price: 230,
        },
        {
          id: 2,
          name: 'child',
          price: 65,
        },
      ],
    },
  ];
};

export const STATIC_SEAT_TYPES = getMockSeatTypes();

const hall = new Hall({
  Seat: SeatNode,
  hallPlanMatrix: HALL_PLAN_MATRIX,
  seatWidth: SEAT_WIDTH,
  seatHeight: SEAT_HEIGHT,
  seatOffset: SEAT_OFFSET,
});

export const SEATS = hall.createSeats();
export const CANVAS_SIZE = hall.getCanvasSize();

export const SEATS_DATA: SeatData[] = SEATS.map((seat) => {
  const { type } = seat;
  const seatType = STATIC_SEAT_TYPES.find((seatType) => seatType.id === type)!;
  const [defaultTicketType] = seatType.ticketTypes;
  const { price, id: ticketTypeId } = defaultTicketType;

  return { ...seat, seatType, ticketTypeId, price };
});
