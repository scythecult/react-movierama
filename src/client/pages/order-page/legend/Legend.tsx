import { SeatStateMap } from '@/client/lib/constants/common';
import type { SeatType } from '@/client/lib/types/OrderPageData';
import { Seat } from '../seats/seat/Seat';
import styles from './styles.module.css';

export type LegendProps = {
  seatTypes: SeatType[];
};

export const Legend = (props: LegendProps) => {
  const { seatTypes } = props;

  const seatTypeNodes = seatTypes.map((seatType) => {
    const { id, name, ticketTypes } = seatType;
    const [defaultTicketType] = ticketTypes;

    return (
      <li className={styles.legendItem} key={id} data-test-id="legend-item">
        <Seat className={styles.legendItemSeat} type={id} />
        <div className={styles.legendItemWrap} data-test-id="legend-item-wrap">
          <small className={styles.legendItemName} data-test-id="legend-item-name">
            {name}
          </small>
          <small className={styles.legendItemPrice} data-test-id="legend-item-price">
            {defaultTicketType.price} RUB
          </small>
        </div>
      </li>
    );
  });

  return (
    <ul className={styles.legend} data-test-id="legend">
      {seatTypeNodes}
      <li className={styles.legendItem} data-test-id="legend-item">
        <Seat className={styles.legendItemSeat} state={SeatStateMap.FREE} />
        <small className={styles.legendItemName} data-test-id="legend-item-name">
          N/A
        </small>
      </li>
      <li className={styles.legendItem} data-test-id="legend-item">
        <Seat className={styles.legendItemSeat} state={SeatStateMap.SELECTED} />
        <small className={styles.legendItemName} data-test-id="legend-item-name">
          Selected
        </small>
      </li>
      <li className={styles.legendItem} data-test-id="legend-item">
        <Seat className={styles.legendItemSeat} state={SeatStateMap.OCCUPIED} />
        <small className={styles.legendItemName} data-test-id="legend-item-name">
          Occupied
        </small>
      </li>
    </ul>
  );
};
