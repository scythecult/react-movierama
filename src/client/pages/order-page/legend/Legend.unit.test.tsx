import { render } from '@testing-library/react';
import { STATIC_SEAT_TYPES } from '../../../lib/utils/mocks';
import { Legend, type LegendProps } from './Legend';

const DEFAULT_PROPS: LegendProps = {
  seatTypes: [],
};

describe('Legend', () => {
  test('should correspond default layout', () => {
    const result = render(<Legend {...DEFAULT_PROPS} />);

    expect(result.container).toMatchSnapshot();
  });

  test('should support the "seatTypes" prop', () => {
    const result = render(<Legend seatTypes={STATIC_SEAT_TYPES} />);
    expect(result.container).toMatchSnapshot();
  });
});
