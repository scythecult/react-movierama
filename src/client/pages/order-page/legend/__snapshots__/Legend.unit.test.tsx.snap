// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Legend > should correspond default layout 1`] = `
<div>
  <ul
    class="_legend_8dc83b"
    data-test-id="legend"
  >
    <li
      class="_legendItem_8dc83b"
      data-test-id="legend-item"
    >
      <button
        class="_seat_35b3f7 _legendItemSeat_8dc83b _regular_35b3f7 _free_35b3f7"
        data-test-disabled="false"
        data-test-id="seat"
        data-test-selected="false"
        type="button"
      >
        <span
          class="_seatPlace_35b3f7"
          data-test-id="seat-place"
        />
        <svg
          class="_seatIcon_35b3f7"
          data-test-id="seat-icon"
          fill="currentColor"
          height="20"
          viewBox="0 0 18 20"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.813 0c-.434 0-1.071.578-1.33.828l-.623.611a.572.572 0 0 1-.402.16.574.574 0 0 1-.402-.16l-.633-.613C12.998.416 12.293 0 11.769 0s-1.229.416-1.654.826l-.634.613a.578.578 0 0 1-.405.16.578.578 0 0 1-.405-.16L8.036.826C7.622.426 6.901 0 6.382 0c-.52 0-1.24.426-1.654.826l-.635.613a.579.579 0 0 1-.405.16.579.579 0 0 1-.405-.16L2.648.826C2.613.793 1.772 0 1.263 0 .683 0 .461.787.461 1.461V18.54c0 .674.221 1.461.802 1.461.49 0 1.279-.742 1.365-.825l.645-.614a.585.585 0 0 1 .409-.16c.155 0 .3.057.408.16l.636.613c.414.4 1.135.826 1.655.826s1.24-.426 1.655-.826l.635-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.414.4 1.135.826 1.654.826.52 0 1.24-.426 1.655-.826l.634-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.32.308.9.826 1.309.826.478 0 .726-.519.726-1.461V1.46c0-.94-.249-1.46-.726-1.46zm-3.018 15.023H9.593v-1.735h4.2v1.735zm0-4.201H4.296V9.178h9.497v1.644zm0-4.11H4.296V4.977h9.497v1.735z"
          />
        </svg>
      </button>
      <small
        class="_legendItemName_8dc83b"
        data-test-id="legend-item-name"
      >
        N/A
      </small>
    </li>
    <li
      class="_legendItem_8dc83b"
      data-test-id="legend-item"
    >
      <button
        class="_seat_35b3f7 _legendItemSeat_8dc83b _regular_35b3f7 _selected_35b3f7"
        data-test-disabled="false"
        data-test-id="seat"
        data-test-selected="true"
        type="button"
      >
        <span
          class="_seatPlace_35b3f7"
          data-test-id="seat-place"
        />
        <svg
          class="_seatIcon_35b3f7"
          data-test-id="seat-icon"
          fill="currentColor"
          height="20"
          viewBox="0 0 18 20"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.813 0c-.434 0-1.071.578-1.33.828l-.623.611a.572.572 0 0 1-.402.16.574.574 0 0 1-.402-.16l-.633-.613C12.998.416 12.293 0 11.769 0s-1.229.416-1.654.826l-.634.613a.578.578 0 0 1-.405.16.578.578 0 0 1-.405-.16L8.036.826C7.622.426 6.901 0 6.382 0c-.52 0-1.24.426-1.654.826l-.635.613a.579.579 0 0 1-.405.16.579.579 0 0 1-.405-.16L2.648.826C2.613.793 1.772 0 1.263 0 .683 0 .461.787.461 1.461V18.54c0 .674.221 1.461.802 1.461.49 0 1.279-.742 1.365-.825l.645-.614a.585.585 0 0 1 .409-.16c.155 0 .3.057.408.16l.636.613c.414.4 1.135.826 1.655.826s1.24-.426 1.655-.826l.635-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.414.4 1.135.826 1.654.826.52 0 1.24-.426 1.655-.826l.634-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.32.308.9.826 1.309.826.478 0 .726-.519.726-1.461V1.46c0-.94-.249-1.46-.726-1.46zm-3.018 15.023H9.593v-1.735h4.2v1.735zm0-4.201H4.296V9.178h9.497v1.644zm0-4.11H4.296V4.977h9.497v1.735z"
          />
        </svg>
      </button>
      <small
        class="_legendItemName_8dc83b"
        data-test-id="legend-item-name"
      >
        Selected
      </small>
    </li>
    <li
      class="_legendItem_8dc83b"
      data-test-id="legend-item"
    >
      <button
        class="_seat_35b3f7 _legendItemSeat_8dc83b _regular_35b3f7 _occupied_35b3f7"
        data-test-disabled="true"
        data-test-id="seat"
        data-test-selected="false"
        disabled=""
        type="button"
      >
        <span
          class="_seatPlace_35b3f7"
          data-test-id="seat-place"
        />
        <svg
          class="_seatIcon_35b3f7"
          data-test-id="seat-icon"
          fill="currentColor"
          height="20"
          viewBox="0 0 18 20"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.813 0c-.434 0-1.071.578-1.33.828l-.623.611a.572.572 0 0 1-.402.16.574.574 0 0 1-.402-.16l-.633-.613C12.998.416 12.293 0 11.769 0s-1.229.416-1.654.826l-.634.613a.578.578 0 0 1-.405.16.578.578 0 0 1-.405-.16L8.036.826C7.622.426 6.901 0 6.382 0c-.52 0-1.24.426-1.654.826l-.635.613a.579.579 0 0 1-.405.16.579.579 0 0 1-.405-.16L2.648.826C2.613.793 1.772 0 1.263 0 .683 0 .461.787.461 1.461V18.54c0 .674.221 1.461.802 1.461.49 0 1.279-.742 1.365-.825l.645-.614a.585.585 0 0 1 .409-.16c.155 0 .3.057.408.16l.636.613c.414.4 1.135.826 1.655.826s1.24-.426 1.655-.826l.635-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.414.4 1.135.826 1.654.826.52 0 1.24-.426 1.655-.826l.634-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.32.308.9.826 1.309.826.478 0 .726-.519.726-1.461V1.46c0-.94-.249-1.46-.726-1.46zm-3.018 15.023H9.593v-1.735h4.2v1.735zm0-4.201H4.296V9.178h9.497v1.644zm0-4.11H4.296V4.977h9.497v1.735z"
          />
        </svg>
      </button>
      <small
        class="_legendItemName_8dc83b"
        data-test-id="legend-item-name"
      >
        Occupied
      </small>
    </li>
  </ul>
</div>
`;

exports[`Legend > should support the "seatTypes" prop 1`] = `
<div>
  <ul
    class="_legend_8dc83b"
    data-test-id="legend"
  >
    <li
      class="_legendItem_8dc83b"
      data-test-id="legend-item"
    >
      <button
        class="_seat_35b3f7 _legendItemSeat_8dc83b _regular_35b3f7 _free_35b3f7"
        data-test-disabled="false"
        data-test-id="seat"
        data-test-selected="false"
        type="button"
      >
        <span
          class="_seatPlace_35b3f7"
          data-test-id="seat-place"
        />
        <svg
          class="_seatIcon_35b3f7"
          data-test-id="seat-icon"
          fill="currentColor"
          height="20"
          viewBox="0 0 18 20"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.813 0c-.434 0-1.071.578-1.33.828l-.623.611a.572.572 0 0 1-.402.16.574.574 0 0 1-.402-.16l-.633-.613C12.998.416 12.293 0 11.769 0s-1.229.416-1.654.826l-.634.613a.578.578 0 0 1-.405.16.578.578 0 0 1-.405-.16L8.036.826C7.622.426 6.901 0 6.382 0c-.52 0-1.24.426-1.654.826l-.635.613a.579.579 0 0 1-.405.16.579.579 0 0 1-.405-.16L2.648.826C2.613.793 1.772 0 1.263 0 .683 0 .461.787.461 1.461V18.54c0 .674.221 1.461.802 1.461.49 0 1.279-.742 1.365-.825l.645-.614a.585.585 0 0 1 .409-.16c.155 0 .3.057.408.16l.636.613c.414.4 1.135.826 1.655.826s1.24-.426 1.655-.826l.635-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.414.4 1.135.826 1.654.826.52 0 1.24-.426 1.655-.826l.634-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.32.308.9.826 1.309.826.478 0 .726-.519.726-1.461V1.46c0-.94-.249-1.46-.726-1.46zm-3.018 15.023H9.593v-1.735h4.2v1.735zm0-4.201H4.296V9.178h9.497v1.644zm0-4.11H4.296V4.977h9.497v1.735z"
          />
        </svg>
      </button>
      <div
        class="_legendItemWrap_8dc83b"
        data-test-id="legend-item-wrap"
      >
        <small
          class="_legendItemName_8dc83b"
          data-test-id="legend-item-name"
        >
          regular
        </small>
        <small
          class="_legendItemPrice_8dc83b"
          data-test-id="legend-item-price"
        >
          330
           RUB
        </small>
      </div>
    </li>
    <li
      class="_legendItem_8dc83b"
      data-test-id="legend-item"
    >
      <button
        class="_seat_35b3f7 _legendItemSeat_8dc83b _comfort_35b3f7 _free_35b3f7"
        data-test-disabled="false"
        data-test-id="seat"
        data-test-selected="false"
        type="button"
      >
        <span
          class="_seatPlace_35b3f7"
          data-test-id="seat-place"
        />
        <svg
          class="_seatIcon_35b3f7"
          data-test-id="seat-icon"
          fill="currentColor"
          height="20"
          viewBox="0 0 18 20"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.813 0c-.434 0-1.071.578-1.33.828l-.623.611a.572.572 0 0 1-.402.16.574.574 0 0 1-.402-.16l-.633-.613C12.998.416 12.293 0 11.769 0s-1.229.416-1.654.826l-.634.613a.578.578 0 0 1-.405.16.578.578 0 0 1-.405-.16L8.036.826C7.622.426 6.901 0 6.382 0c-.52 0-1.24.426-1.654.826l-.635.613a.579.579 0 0 1-.405.16.579.579 0 0 1-.405-.16L2.648.826C2.613.793 1.772 0 1.263 0 .683 0 .461.787.461 1.461V18.54c0 .674.221 1.461.802 1.461.49 0 1.279-.742 1.365-.825l.645-.614a.585.585 0 0 1 .409-.16c.155 0 .3.057.408.16l.636.613c.414.4 1.135.826 1.655.826s1.24-.426 1.655-.826l.635-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.414.4 1.135.826 1.654.826.52 0 1.24-.426 1.655-.826l.634-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.32.308.9.826 1.309.826.478 0 .726-.519.726-1.461V1.46c0-.94-.249-1.46-.726-1.46zm-3.018 15.023H9.593v-1.735h4.2v1.735zm0-4.201H4.296V9.178h9.497v1.644zm0-4.11H4.296V4.977h9.497v1.735z"
          />
        </svg>
      </button>
      <div
        class="_legendItemWrap_8dc83b"
        data-test-id="legend-item-wrap"
      >
        <small
          class="_legendItemName_8dc83b"
          data-test-id="legend-item-name"
        >
          comfort
        </small>
        <small
          class="_legendItemPrice_8dc83b"
          data-test-id="legend-item-price"
        >
          380
           RUB
        </small>
      </div>
    </li>
    <li
      class="_legendItem_8dc83b"
      data-test-id="legend-item"
    >
      <button
        class="_seat_35b3f7 _legendItemSeat_8dc83b _vip_35b3f7 _free_35b3f7"
        data-test-disabled="false"
        data-test-id="seat"
        data-test-selected="false"
        type="button"
      >
        <span
          class="_seatPlace_35b3f7"
          data-test-id="seat-place"
        />
        <svg
          class="_seatIcon_35b3f7"
          data-test-id="seat-icon"
          fill="currentColor"
          height="20"
          viewBox="0 0 18 20"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.813 0c-.434 0-1.071.578-1.33.828l-.623.611a.572.572 0 0 1-.402.16.574.574 0 0 1-.402-.16l-.633-.613C12.998.416 12.293 0 11.769 0s-1.229.416-1.654.826l-.634.613a.578.578 0 0 1-.405.16.578.578 0 0 1-.405-.16L8.036.826C7.622.426 6.901 0 6.382 0c-.52 0-1.24.426-1.654.826l-.635.613a.579.579 0 0 1-.405.16.579.579 0 0 1-.405-.16L2.648.826C2.613.793 1.772 0 1.263 0 .683 0 .461.787.461 1.461V18.54c0 .674.221 1.461.802 1.461.49 0 1.279-.742 1.365-.825l.645-.614a.585.585 0 0 1 .409-.16c.155 0 .3.057.408.16l.636.613c.414.4 1.135.826 1.655.826s1.24-.426 1.655-.826l.635-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.414.4 1.135.826 1.654.826.52 0 1.24-.426 1.655-.826l.634-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.32.308.9.826 1.309.826.478 0 .726-.519.726-1.461V1.46c0-.94-.249-1.46-.726-1.46zm-3.018 15.023H9.593v-1.735h4.2v1.735zm0-4.201H4.296V9.178h9.497v1.644zm0-4.11H4.296V4.977h9.497v1.735z"
          />
        </svg>
      </button>
      <div
        class="_legendItemWrap_8dc83b"
        data-test-id="legend-item-wrap"
      >
        <small
          class="_legendItemName_8dc83b"
          data-test-id="legend-item-name"
        >
          vip
        </small>
        <small
          class="_legendItemPrice_8dc83b"
          data-test-id="legend-item-price"
        >
          430
           RUB
        </small>
      </div>
    </li>
    <li
      class="_legendItem_8dc83b"
      data-test-id="legend-item"
    >
      <button
        class="_seat_35b3f7 _legendItemSeat_8dc83b _wheelchair_35b3f7 _free_35b3f7"
        data-test-disabled="false"
        data-test-id="seat"
        data-test-selected="false"
        type="button"
      >
        <span
          class="_seatPlace_35b3f7"
          data-test-id="seat-place"
        />
        <svg
          class="_seatIcon_35b3f7"
          data-test-id="seat-icon"
          fill="currentColor"
          height="20"
          viewBox="0 0 18 20"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.813 0c-.434 0-1.071.578-1.33.828l-.623.611a.572.572 0 0 1-.402.16.574.574 0 0 1-.402-.16l-.633-.613C12.998.416 12.293 0 11.769 0s-1.229.416-1.654.826l-.634.613a.578.578 0 0 1-.405.16.578.578 0 0 1-.405-.16L8.036.826C7.622.426 6.901 0 6.382 0c-.52 0-1.24.426-1.654.826l-.635.613a.579.579 0 0 1-.405.16.579.579 0 0 1-.405-.16L2.648.826C2.613.793 1.772 0 1.263 0 .683 0 .461.787.461 1.461V18.54c0 .674.221 1.461.802 1.461.49 0 1.279-.742 1.365-.825l.645-.614a.585.585 0 0 1 .409-.16c.155 0 .3.057.408.16l.636.613c.414.4 1.135.826 1.655.826s1.24-.426 1.655-.826l.635-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.414.4 1.135.826 1.654.826.52 0 1.24-.426 1.655-.826l.634-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.32.308.9.826 1.309.826.478 0 .726-.519.726-1.461V1.46c0-.94-.249-1.46-.726-1.46zm-3.018 15.023H9.593v-1.735h4.2v1.735zm0-4.201H4.296V9.178h9.497v1.644zm0-4.11H4.296V4.977h9.497v1.735z"
          />
        </svg>
      </button>
      <div
        class="_legendItemWrap_8dc83b"
        data-test-id="legend-item-wrap"
      >
        <small
          class="_legendItemName_8dc83b"
          data-test-id="legend-item-name"
        >
          wheelchair
        </small>
        <small
          class="_legendItemPrice_8dc83b"
          data-test-id="legend-item-price"
        >
          230
           RUB
        </small>
      </div>
    </li>
    <li
      class="_legendItem_8dc83b"
      data-test-id="legend-item"
    >
      <button
        class="_seat_35b3f7 _legendItemSeat_8dc83b _regular_35b3f7 _free_35b3f7"
        data-test-disabled="false"
        data-test-id="seat"
        data-test-selected="false"
        type="button"
      >
        <span
          class="_seatPlace_35b3f7"
          data-test-id="seat-place"
        />
        <svg
          class="_seatIcon_35b3f7"
          data-test-id="seat-icon"
          fill="currentColor"
          height="20"
          viewBox="0 0 18 20"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.813 0c-.434 0-1.071.578-1.33.828l-.623.611a.572.572 0 0 1-.402.16.574.574 0 0 1-.402-.16l-.633-.613C12.998.416 12.293 0 11.769 0s-1.229.416-1.654.826l-.634.613a.578.578 0 0 1-.405.16.578.578 0 0 1-.405-.16L8.036.826C7.622.426 6.901 0 6.382 0c-.52 0-1.24.426-1.654.826l-.635.613a.579.579 0 0 1-.405.16.579.579 0 0 1-.405-.16L2.648.826C2.613.793 1.772 0 1.263 0 .683 0 .461.787.461 1.461V18.54c0 .674.221 1.461.802 1.461.49 0 1.279-.742 1.365-.825l.645-.614a.585.585 0 0 1 .409-.16c.155 0 .3.057.408.16l.636.613c.414.4 1.135.826 1.655.826s1.24-.426 1.655-.826l.635-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.414.4 1.135.826 1.654.826.52 0 1.24-.426 1.655-.826l.634-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.32.308.9.826 1.309.826.478 0 .726-.519.726-1.461V1.46c0-.94-.249-1.46-.726-1.46zm-3.018 15.023H9.593v-1.735h4.2v1.735zm0-4.201H4.296V9.178h9.497v1.644zm0-4.11H4.296V4.977h9.497v1.735z"
          />
        </svg>
      </button>
      <small
        class="_legendItemName_8dc83b"
        data-test-id="legend-item-name"
      >
        N/A
      </small>
    </li>
    <li
      class="_legendItem_8dc83b"
      data-test-id="legend-item"
    >
      <button
        class="_seat_35b3f7 _legendItemSeat_8dc83b _regular_35b3f7 _selected_35b3f7"
        data-test-disabled="false"
        data-test-id="seat"
        data-test-selected="true"
        type="button"
      >
        <span
          class="_seatPlace_35b3f7"
          data-test-id="seat-place"
        />
        <svg
          class="_seatIcon_35b3f7"
          data-test-id="seat-icon"
          fill="currentColor"
          height="20"
          viewBox="0 0 18 20"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.813 0c-.434 0-1.071.578-1.33.828l-.623.611a.572.572 0 0 1-.402.16.574.574 0 0 1-.402-.16l-.633-.613C12.998.416 12.293 0 11.769 0s-1.229.416-1.654.826l-.634.613a.578.578 0 0 1-.405.16.578.578 0 0 1-.405-.16L8.036.826C7.622.426 6.901 0 6.382 0c-.52 0-1.24.426-1.654.826l-.635.613a.579.579 0 0 1-.405.16.579.579 0 0 1-.405-.16L2.648.826C2.613.793 1.772 0 1.263 0 .683 0 .461.787.461 1.461V18.54c0 .674.221 1.461.802 1.461.49 0 1.279-.742 1.365-.825l.645-.614a.585.585 0 0 1 .409-.16c.155 0 .3.057.408.16l.636.613c.414.4 1.135.826 1.655.826s1.24-.426 1.655-.826l.635-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.414.4 1.135.826 1.654.826.52 0 1.24-.426 1.655-.826l.634-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.32.308.9.826 1.309.826.478 0 .726-.519.726-1.461V1.46c0-.94-.249-1.46-.726-1.46zm-3.018 15.023H9.593v-1.735h4.2v1.735zm0-4.201H4.296V9.178h9.497v1.644zm0-4.11H4.296V4.977h9.497v1.735z"
          />
        </svg>
      </button>
      <small
        class="_legendItemName_8dc83b"
        data-test-id="legend-item-name"
      >
        Selected
      </small>
    </li>
    <li
      class="_legendItem_8dc83b"
      data-test-id="legend-item"
    >
      <button
        class="_seat_35b3f7 _legendItemSeat_8dc83b _regular_35b3f7 _occupied_35b3f7"
        data-test-disabled="true"
        data-test-id="seat"
        data-test-selected="false"
        disabled=""
        type="button"
      >
        <span
          class="_seatPlace_35b3f7"
          data-test-id="seat-place"
        />
        <svg
          class="_seatIcon_35b3f7"
          data-test-id="seat-icon"
          fill="currentColor"
          height="20"
          viewBox="0 0 18 20"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.813 0c-.434 0-1.071.578-1.33.828l-.623.611a.572.572 0 0 1-.402.16.574.574 0 0 1-.402-.16l-.633-.613C12.998.416 12.293 0 11.769 0s-1.229.416-1.654.826l-.634.613a.578.578 0 0 1-.405.16.578.578 0 0 1-.405-.16L8.036.826C7.622.426 6.901 0 6.382 0c-.52 0-1.24.426-1.654.826l-.635.613a.579.579 0 0 1-.405.16.579.579 0 0 1-.405-.16L2.648.826C2.613.793 1.772 0 1.263 0 .683 0 .461.787.461 1.461V18.54c0 .674.221 1.461.802 1.461.49 0 1.279-.742 1.365-.825l.645-.614a.585.585 0 0 1 .409-.16c.155 0 .3.057.408.16l.636.613c.414.4 1.135.826 1.655.826s1.24-.426 1.655-.826l.635-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.414.4 1.135.826 1.654.826.52 0 1.24-.426 1.655-.826l.634-.613a.578.578 0 0 1 .405-.16c.155 0 .299.057.405.16l.635.613c.32.308.9.826 1.309.826.478 0 .726-.519.726-1.461V1.46c0-.94-.249-1.46-.726-1.46zm-3.018 15.023H9.593v-1.735h4.2v1.735zm0-4.201H4.296V9.178h9.497v1.644zm0-4.11H4.296V4.977h9.497v1.735z"
          />
        </svg>
      </button>
      <small
        class="_legendItemName_8dc83b"
        data-test-id="legend-item-name"
      >
        Occupied
      </small>
    </li>
  </ul>
</div>
`;
